analyses/survival-analysis/util/survival_models.R

```diff
@@ -0,0 +1,117 @@
+# Functions for conducting survival analyses
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2019
+#
+# Install the packages we need
+if (!("survival" %in% installed.packages())) {
+  install.packages("survival")
+}
+
+if (!("survminer" %in% installed.packages())) {
+  install.packages("survminer")
+}
+
+# Attach this package
+library(survminer)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+survival_analysis <- function(metadata, ind_var, test = "kap.meier", ind_data = NULL) {
+  # Given the overall survival information, and an independent variable will
+  # run survival analysis and return a list with 1) the model fit object and
+  # 2) a summary table 3) the data used.
+  #
+  # Args:
+  #   metadata: a data.frame that contains columns OS_status and OS_days to use
+  #             for the survival model. This also assumes "LIVING" and "DECEASED"
+  #             are the two statuses. THis will be converted to a numeric variable
+  #             for use with `survival` R functions. Samples with NAs are dropped.
+  #   ind_var: a character string noting the name of the independent variable to
+  #            test as a predictor for survival.
+  #   test: a character string noting which test to use. Supported choices:
+  #         "kap.meier", "log.rank", "cox.reg"
+  #   ind_data: If the data for the independent variable needed for the test is not
+  #         in metadata, add it here. This assumes it will be a data.frame with
+  #         either a "Tumor_Sample_Barcode", "Kids_First_Biospecimen_ID", or
+  #         "sample" column which to inner_join with metadata by.
+  #
+  # Returns:
+  # A list with three objects: 1) the original model fit object,
+  #                            2) the summary table
+  #                            3) the original data.frame
+
+  # List the tests
+  supported_tests <- c("kap.meier", "log.rank", "cox.reg")
+
+  # Check that it is a supported test
+  if (!(test %in% supported_tests)) {
+    stop(paste0(test, "is not a supported test."))
+  }
+
+  # Format the OS_status variable for the survival model functions
+  metadata$OS_status <- factor(metadata$OS_status, levels = c("LIVING", "DECEASED"))
```

Comment:
This feels like a very strong assumption, to require specific column names in `metadata`. A more flexible solution will use tidyeval where users can provide the column names containing this information. I'd recommend using named arguments for this.
Depending on the `rlang` version we have associated with this project's docker image, there are two solutions:
1. If `rlang >=0.40`: [https://rdrr.io/github/tidyverse/rlang/f/NEWS.md](https://rdrr.io/github/tidyverse/rlang/f/NEWS.md). See section entitled "Interpolate function inputs with the curly-curly operator"
2. If earlier `rlang`, it's more annoying but still very doable (see bits about enquo) [https://tidyeval.tidyverse.org/dplyr.html](https://tidyeval.tidyverse.org/dplyr.html)

Reply:
I agree that this is a strong assumption if we were using this for other data, but for the given project and metadata file I don't *think* these column names will be changing. That being said, I'm all for making things more generalizable/flexible so I'll checkout these sources!